pkg/virt-controller/watch/descheduler/descheduler.go

```diff
@@ -19,7 +19,85 @@
 
 package descheduler
 
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/client-go/tools/cache"
+	virtv1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/controller"
+
+	k8sv1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/client-go/log"
+
+	"kubevirt.io/kubevirt/pkg/apimachinery/patch"
+)
+
 // EvictOnlyAnnotation indicates pods whose eviction is not expected to be completed right away.
 // Instead, an eviction request is expected to be intercepted by an external component which will initiate the
 // eviction process for the pod.
 const EvictOnlyAnnotation = "descheduler.alpha.kubernetes.io/request-evict-only"
+
+// EvictionInProgressAnnotation indicates pods whose eviction was initiated by an external component.
+const EvictionInProgressAnnotation = "descheduler.alpha.kubernetes.io/eviction-in-progress"
+
+func AddDeschedulerAnnotation(virtClient kubecli.KubevirtClient, sourcePod *k8sv1.Pod) error {
+	if _, exists := sourcePod.GetAnnotations()[EvictionInProgressAnnotation]; exists {
+		return nil
+	}
+
+	patchSet := patch.New(
+		patch.WithAdd(fmt.Sprintf("/metadata/annotations/%s", patch.EscapeJSONPointer(EvictionInProgressAnnotation)), ""),
```

Comment:
What happens if it exists already? Maybe a replace works better? The check you do before the patch doesn't guarantee 100% that the annotation won't exist as the Informer store might miss an update.

Reply:
:+1:

Reply:
Actually, seems like `add` is better than `replace` https://jsonpatch.com/#replace.
But tested locally they work the same...

Reply:
I just want to make sure it doesn't fail if the annotation is already there, does it?

Reply:
Nope, it does not. I tried with both kubectl and API request and everything is working.
i.e. This test:
```
			vmi := libvmifact.NewCirros()
			vmi = tests.RunVMIAndExpectLaunch(vmi, 60)
			patchBytes, err := patch.New(patch.WithAdd("/metadata/annotations/test", "false")).GeneratePayload()
			Expect(err).ToNot(HaveOccurred())

			vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Patch(context.Background(), vmi.Name, types.JSONPatchType, patchBytes, metav1.PatchOptions{})
			Expect(err).ToNot(HaveOccurred())

			patchBytes, err = patch.New(patch.WithAdd("/metadata/annotations/test", "last")).GeneratePayload()
			Expect(err).ToNot(HaveOccurred())

			vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Patch(context.Background(), vmi.Name, types.JSONPatchType, patchBytes, metav1.PatchOptions{})
			Expect(err).ToNot(HaveOccurred())
```
passes.